recipes/poselib/all/conanfile.py

```diff
@@ -0,0 +1,108 @@
+import os
+
+from conan import ConanFile
+from conan.errors import ConanInvalidConfiguration
+from conan.tools.build import check_min_cppstd
+from conan.tools.cmake import CMake, CMakeDeps, CMakeToolchain, cmake_layout
+from conan.tools.files import copy, get, replace_in_file, rmdir
+from conan.tools.scm import Version
+
+required_conan_version = ">=1.53.0"
+
+
+class PoselibConan(ConanFile):
+    name = "poselib"
+    description = "PoseLib: minimal solvers for calibrated camera pose estimation"
+    license = "BSD-3-Clause"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/PoseLib/PoseLib"
+    topics = ("pose", "camera", "estimation", "solver")
+
+    package_type = "library"
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+    }
+
+    @property
+    def _min_cppstd(self):
+        return 17
+
+    @property
+    def _compilers_minimum_version(self):
+        return {
+            "gcc": "7",
+            "clang": "6",
+            "apple-clang": "10",
+            "msvc": "192",
+            "Visual Studio": "16",
+        }
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            self.options.rm_safe("fPIC")
+
+    def layout(self):
+        cmake_layout(self, src_folder="src")
+
+    def requirements(self):
+        self.requires("eigen/3.4.0", transitive_headers=True, transitive_libs=True)
+
+    def validate(self):
+        if self.settings.compiler.cppstd:
+            check_min_cppstd(self, self._min_cppstd)
+        minimum_version = self._compilers_minimum_version.get(str(self.settings.compiler), False)
+        if minimum_version and Version(self.settings.compiler.version) < minimum_version:
+            raise ConanInvalidConfiguration(
+                f"{self.ref} requires C++{self._min_cppstd}, which your compiler does not support."
+            )
+
+        if self.settings.os == "Windows" and self.options.shared:
+            # Symbols are not exported
+            raise ConanInvalidConfiguration("Windows shared builds are not supported")
+
+    def source(self):
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
+
+    def generate(self):
+        tc = CMakeToolchain(self)
+        tc.cache_variables["CMAKE_POLICY_DEFAULT_CMP0077"] = "NEW"
```

Comment:
As we are not defining any variable, and following [this](https://github.com/conan-io/conan-center-index/pull/22064#discussion_r1695188168) comment, I think for now there is no need to define this policy!

Reply:
It still applies to the variables set automatically, such as `BUILD_SHARED_LIBS`, which are configured via `option()` in the project.

Reply:
Confirmed: https://github.com/PoseLib/PoseLib/blob/90f47ffce946569f00f410a528a6a7c4fce5fea2/cmake/SetEnv.cmake#L20